array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,21 @@
+/* exported zip */
+function zip(first, second){
+    var output = [];
+    if(first.length > second.length) {
+        for(var i = 0; i < second.length ; i++) {
+            var subArray = [];
+            subArray.push(first[i])
+            subArray.push(second[i]);
+            output.push(subArray);
+        }
+    }
+    else if(second.length >= first.length) {
+        for(var i = 0; i < first.length ; i++) {
+            var subArray = [];
+            subArray.push(first[i])
+            subArray.push(second[i]);
+            output.push(subArray);
+        }
+    }
```

Comment:
If you decide which array is longer before the loop, then you can write the loop only once.

```js
var length = Math.min(first.length, second.length);
```

Reply:
THANK YOU for all the feedback!